pkg/transport/generic_transport_controller.go

```diff
@@ -410,44 +465,174 @@ func (c *genericTransportController) getObjectsFromWDS(ctx context.Context, bind
 	return objectsToPropagate, nil
 }
 
-func (c *genericTransportController) initializeWrappedObject(ctx context.Context, binding *v1alpha1.Binding) (*unstructured.Unstructured, error) {
+func (c *genericTransportController) initializeWrappedObject(ctx context.Context, binding *v1alpha1.Binding) (func(v1alpha1.Destination) (*unstructured.Unstructured, bool), []string, error) {
 	objectsToPropagate, err := c.getObjectsFromWDS(ctx, binding)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get objects to propagate to WECs from Binding object '%s' - %w", binding.GetName(), err)
+		return nil, nil, fmt.Errorf("failed to get objects to propagate to WECs from Binding object '%s' - %w", binding.GetName(), err)
 	}
 
 	if len(objectsToPropagate) == 0 {
-		return nil, nil // if no objects were found in the workload section, return nil so that we don't distribute an empty wrapped object.
+		return nil, nil, nil // if no objects were found in the workload section, return nil so that we don't distribute an empty wrapped object.
+	}
+
+	bindingErrors := []string{}
+	destToDefs := map[v1alpha1.Destination]map[string]string{}
+
+	// This will become non-nil if any object to propagate needs customization
+	var destToCustomizedObjects map[v1alpha1.Destination][]*unstructured.Unstructured
+
+	// Look through the objects to propagate to see if any needs customization.
+	// If any needs customization then catch up destToCustomizedObjects and proceed from there.
+	for objIdx, objToPropagate := range objectsToPropagate {
```

Comment:
The current draft has a different form of simplicity: the distinction between objects that request template expansion and those that do not is concentrated in a narrower part of the code, just the internals of one method (initializeWrappedObject).

The current draft has a different form of efficiency. Iterating over the slice of objects twice instead of once makes a smaller difference than doing unnecessary template expansions. The current draft realizes the opportunity to recognize that an object that requests template expansion does not actually use it. I think that this can happen for any of three reasons. Remember that when KubeStellar requires its user to put an annotation or label on an object coming from a third party, this is a problem. One way to solve this would be to add a boolean in BindingPolicySpec that says to request template expansion on all matching objects. Another would be introduce another control object that has rules identifying the workload objects to be subject to template expansion. Another would be to use a feature of the facility importing the workload objects to add an annotation to all imported objects. In all of these, it is likely that the request paint would be applied with a broad brush.